extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/NettyProcessor.java

```diff
@@ -137,4 +138,10 @@ void createExecutors(BuildProducer<RuntimeBeanBuildItem> runtimeBeanBuildItemBui
                 .build());
     }
 
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
+    void setJBossInternalLoggerFactory() {
```

Comment:
This was not what I meant (and should fail as it is not valid to have a @Record method that does not inject a recorder).

I just meant a static block to make sure it is run when this class is loaded:

  static {
    InternalLoggerFactory.setDefaultFactory(new JBossNettyLoggerFactory());
  }

Anything in a build step has the possibility that other build steps will be run first.

Reply:
> This was not what I meant (and should fail as it is not valid to have a @record method that does not inject a recorder).

Ok... 🙈

There's [an example](https://github.com/quarkusio/quarkus/blob/master/core/test-extension/deployment/src/main/java/io/quarkus/extest/deployment/TestProcessor.java#L231) in the codebase of this.